client/src/app/components/Loader.js

```diff
@@ -1,21 +1,46 @@
 import React, {Component} from "react";
+import {Transition} from "react-transition-group";
 import PropTypes from "prop-types";
 
-import loaderImg from "../../imgs/loaderImg.svg";
 import "../scss/loader.css";
 
 class Loader extends Component {
+
+    constructor(props) {
+        super(props);
+        this.transparent = false;
+    }
+    shouldComponentUpdate(nextProps) {
+        if (nextProps.in !== this.props.in)
+            return true;
+        return false;
+    }
+    componentWillMount() {
+        if (this.props.in) {
+            this.transparent = this.props.transparent;
+        }
+    }
+    componentWillReceiveProps(nextProps) {
+        
+        if (nextProps.in) {
+            this.transparent = nextProps.transparent;
+        }
+    }
     render() {
         return (
-            <div className={this.props.showHover && "loaderHover"}>
-                <img src={loaderImg} className={"loaderImg"} alt={"Loader"} />
-            </div>
+            <Transition in={this.props.in} timeout={0}>
+                {(status) => (
+                    <div className={`${this.transparent ? "transparent" : ""} ${this.props.name ? `loader${this.props.name}` : ""} loader ${status}`} />
```

Comment:
Tu peux directement utiliser `this.props.transparent` et utiliser `Loader.defaultProps = {transparent: false}`

Reply:
la classe transparemt ne doit pas s'enlever quand on hide le loader sinon ca casse les animations. Du coup j'update la variable uniquement au lancement du loader, ce que tu me propose ferra pas ca je crois